build_test.go

```diff
@@ -68,49 +61,106 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
-			it.Before(func() {
-				pyProjectParser.ParseCall.Returns.Error = fmt.Errorf("some error")
-			})
+	context("with BP_POETRY_RUN_TARGET not set", func() {
+		it("returns a result that sets the 'poetry run' launch command", func() {
+			result, err := build(buildContext)
+			Expect(err).NotTo(HaveOccurred())
 
-			it("returns the error", func() {
-				_, err := build(packit.BuildContext{
-					WorkingDir: workingDir,
-					CNBPath:    cnbDir,
-					Stack:      "some-stack",
-					BuildpackInfo: packit.BuildpackInfo{
-						Name:    "Some Buildpack",
-						Version: "some-version",
-					},
-					Plan: packit.BuildpackPlan{
-						Entries: []packit.BuildpackPlanEntry{},
+			Expect(result).To(Equal(packit.BuildResult{
+				Plan: packit.BuildpackPlan{
+					Entries: nil,
+				},
+				Layers: nil,
+				Launch: packit.LaunchMetadata{
+					Processes: []packit.Process{
+						{
+							Type:    "web",
+							Command: "poetry run some-script",
+							Default: true,
+						},
 					},
-					Layers: packit.Layers{Path: layersDir},
+				},
+			}))
+
+			Expect(buffer.String()).To(ContainLines(
+				ContainSubstring("Finding the poetry run target"),
+				ContainSubstring("Found pyproject.toml script=some-script"),
+				ContainSubstring("Assigning launch process"),
+				ContainSubstring("web: poetry run some-script"),
+			))
+		})
+
+		context("failure cases", func() {
+			context("when the pyproject.toml parser returns an error", func() {
+				it.Before(func() {
+					pyProjectParser.ParseCall.Returns.Error = fmt.Errorf("some error")
+				})
+
+				it("returns the error", func() {
+					_, err := build(packit.BuildContext{
+						WorkingDir: workingDir,
+						CNBPath:    cnbDir,
+						Stack:      "some-stack",
+						BuildpackInfo: packit.BuildpackInfo{
+							Name:    "Some Buildpack",
+							Version: "some-version",
+						},
+						Plan: packit.BuildpackPlan{
+							Entries: []packit.BuildpackPlanEntry{},
+						},
+						Layers: packit.Layers{Path: layersDir},
+					})
+					Expect(err).To(MatchError(ContainSubstring("some error")))
 				})
-				Expect(err).To(MatchError(ContainSubstring("some error")))
 			})
 		})
 	})
 
+	context("with BP_POETRY_RUN_TARGET set", func() {
+		it.Before(func() {
+			Expect(os.Setenv("BP_POETRY_RUN_TARGET", "a custom command")).To(Succeed())
+		})
+
+		it.After(func() {
+			Expect(os.Unsetenv("BP_POETRY_RUN_TARGET")).To(Succeed())
+		})
+
+		it("will use the value of BP_POETRY_RUN_TARGET and not use the pyproject.toml parser", func() {
+			result, err := build(buildContext)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(result).To(Equal(packit.BuildResult{
+				Plan: packit.BuildpackPlan{
+					Entries: nil,
+				},
+				Layers: nil,
+				Launch: packit.LaunchMetadata{
+					Processes: []packit.Process{
+						{
+							Type:    "web",
+							Command: "poetry run a custom command",
+							Default: true,
+						},
+					},
+				},
+			}))
+
+			Expect(buffer.String()).To(ContainLines(
```

Comment:
As above, personally I don't think there is much value in unit testing the log lines. The assertions on the tests are sufficient to determine whether we are entering the correct path.